### ISSUE TITLE
Missing JSDoc not reported for generic object-type parameters in TypeScript

### ISSUE THREAD

**Issue by ComradeVanti** [gajus/eslint-plugin-jsdoc#1186]
## Expected behavior

I have the following export in my TypeScript project:

```ts
/**
 * Command options for the login command
 */
export type LoginOptions = CmdOptions<{
  username?: string;
  password?: string;
  email?: string;
  basicAuth?: boolean;
  alwaysAuth?: boolean;
}>;
```

I expect to get a warning that JSDoc for properties is missing.

## Actual behavior

I do not get a warning for the properties.

## Notes

The properties get the expected warning if either:

- I remove the JSDoc above the type declaration
- I make the type a plain object type, ie. remove CmdOption< >

## ESLint Config

```js
module.exports = {
  root: true,
  env: {
    node: true,
  },
  extends: [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:jsdoc/recommended-typescript",
    "prettier",
  ],
  rules: {
    "no-console": process.env.NODE_ENV === "production" ? "error" : "off",
    "no-debugger": process.env.NODE_ENV === "production" ? "error" : "off",
    "jsdoc/require-jsdoc": [
      "warn",
      {
        publicOnly: { ancestorsOnly: true, cjs: true },
        contexts: [
          "TSTypeAliasDeclaration",
          "TSInterfaceDeclaration",
          "TSMethodSignature",
          "TSPropertySignature",
        ],
      },
    ],
    "jsdoc/require-throws": 1,
    "jsdoc/require-returns": 0,
  },
  parser: "@typescript-eslint/parser",
  plugins: ["@typescript-eslint", "jsdoc"],
};
```

## Environment

- Node version: 16.20.2
- ESLint version: 8.51.0
- TypeScript: 5.2.2
- `eslint-plugin-jsdoc` version: 47.0.2
